engineio/src/asynchronous/async_transports/polling.rs

```diff
@@ -80,14 +81,21 @@ impl AsyncTransport for PollingTransport {
         Ok(())
     }
 
-    async fn poll(&self) -> Result<Bytes> {
-        Ok(self
-            .client
-            .get(self.address().await?)
-            .send()
-            .await?
-            .bytes()
-            .await?)
+    async fn stream(&self) -> Result<Pin<Box<dyn Stream<Item = Result<Bytes>> + '_>>> {
+        let address = self.address().await?;
```

Comment:
suggestion: turn this into a stream, then map it to be a stream of streams then flatten rather than awaiting

Reply:
Good catch!!